ledger/db/src/test_utils/mock_ledger.rs

```diff
@@ -300,11 +301,139 @@ pub fn get_test_ledger_blocks(n_blocks: usize) -> Vec<(Block, BlockContents)> {
     blocks_and_contents
 }
 
+/// Get blocks with custom content in order to simulate conditions seen in
+/// production
+///
+/// * `outputs_per_recipient_per_block` - transaction outputs per account per
+///   block
+/// * `num_accounts` - number of accounts in the blocks
+/// * `num_blocks` - number of simulated blocks to create
+/// * `key_images_per_block` - number of key images per block
+/// * `max_token_id` - number of distinct token ids in blocks
+pub fn get_custom_test_ledger_blocks(
+    outputs_per_recipient_per_block: usize,
+    num_accounts: usize,
+    num_blocks: usize,
+    key_images_per_block: usize,
+    max_token_id: u64,
+) -> Vec<(Block, BlockContents)> {
+    let mut rng: StdRng = SeedableRng::from_seed([1u8; 32]);
+
+    // Number of total tx outputs in all blocks
+    let num_outputs: u64 =
+        (num_accounts * outputs_per_recipient_per_block * num_blocks * (max_token_id as usize + 1))
+            as u64;
+    assert!(num_outputs >= 16);
+
+    // Initialize other defaults
+    let picomob_per_output: u64 = (TOTAL_MOB / num_outputs) * 1_000_000_000_000;
```

Comment:
This looks like it's trying to maximize the MOB for the blocks.  If that is really the case should this logic be reworked to maximize it, something like:

```suggestion
    let picomob_per_output: u64 = (TOTAL_MOB * 1_000_000_000_000) / num_outputs);
```

As it currently stands it will be truncating multiples of `1_000_000_000_000`

Reply:
(TOTAL_MOB * 1_000_000_000_000) would be an integer overflow, but I see what you're suggesting. 

This code just tries to make all outputs equal. There will be some truncation anyways, and that's okay because this test function is meant for creating content-full blocks for benchmark testing. The danger of how it currently is that truncates to zero if num_ouputs > TOTAL_MOB. I added a quick change to address it.

Reply:
```suggestion
    let picomob_per_output: u64 = ((TOTAL_MOB as f64 / num_outputs as f64) * 1000000000000.0) as u64;
```